### PR TITLE
lite mode support

### PR DESCRIPTION
**Pull request by GalaxySciTech**
Adapt Subnet UI for lite relayer
Show 'No funds' or 'Low funds' in relayer status UI if wallet empty

### CODE REVIEW

File: backend/src/services/block.service.ts [hunk 1]
```diff
@@ -206,16 +207,22 @@ export class BlockService {
...
     const { smartContractHeight, smartContractCommittedHash } = await this.getAndSetLastSubmittedBlockInfo();
+    const mode = await this.parentChainClient.mode();
     const { timestamp } = await this.parentChainClient.getParentChainBlockBySubnetHash(smartContractCommittedHash);
     let isProcessing = true;
     const timeDiff = new Date().getTime() / 1000 - parseInt(timestamp.toString());
-    if (timeDiff > 60) isProcessing = false;
+    if (mode == 'lite') {
+      if (timeDiff > 1000) isProcessing = false;
+    } else {
+      if (timeDiff > 60) isProcessing = false;
```
Comment (wanwiset25) [inline backend/src/services/block.service.ts:219]: can we use 120, put some buffer here

Comment (GalaxySciTech) [reply backend/src/services/block.service.ts]: ok

**Review by wanwiset25 — DISMISSED** [review]

File: backend/src/client/parentchain/index.ts [hunk 2]
```diff
@@ -58,6 +58,16 @@ export class ParentChainClient {
     }
   }
 
+  async mode(): Promise<string> {
+    try {
+      const result = await this.smartContractInstance.methods.MODE().call();
+      return result;
+    } catch (error) {
+      logger.error("Error while trying to fetch the last audited subnet's block in XDC mainnet", { message: error.message });
```
Comment (wjrjerome) [inline backend/src/client/parentchain/index.ts:66]: is the error msg correct?

File: backend/src/services/block.service.ts [hunk 3]
```diff
@@ -135,14 +135,15 @@ export class BlockService {
   }
 
   // Returns the block num difference between what's mined in subnet and what's submitted to parent chain
-  public async getProcessingBacklog(): Promise<{ gap: number; isProcessing: boolean }> {
+  public async getProcessingBacklog(): Promise<{ gap: number; isProcessing: boolean; mode: string }> {
```
Comment (wjrjerome) [inline backend/src/services/block.service.ts:138]: is mode an enum?

File: backend/src/services/block.service.ts [hunk 4]
```diff
@@ -206,16 +207,22 @@ export class BlockService {
...
 
-  private async getSmartContractProcessingInfo(): Promise<{ processedUntil: number; isProcessing: boolean }> {
+  private async getSmartContractProcessingInfo(): Promise<{ processedUntil: number; isProcessing: boolean; mode: string }> {
     const { smartContractHeight, smartContractCommittedHash } = await this.getAndSetLastSubmittedBlockInfo();
+    const mode = await this.parentChainClient.mode();
     const { timestamp } = await this.parentChainClient.getParentChainBlockBySubnetHash(smartContractCommittedHash);
     let isProcessing = true;
     const timeDiff = new Date().getTime() / 1000 - parseInt(timestamp.toString());
-    if (timeDiff > 60) isProcessing = false;
+    if (mode == 'lite') {
```
Comment (wjrjerome) [inline backend/src/services/block.service.ts:216]: const isProcessing = timeDiff < 120 || (mode == 'lite' && timeDiff<1000)

Comment (GalaxySciTech) [reply backend/src/services/block.service.ts:216]: the line mean is 
only full ,check isProcessing is timeDiff < 120 will be  true , or false
only lite ,check isProcessing is timeDiff < 1000 will be  true ,or false

**Review by wjrjerome — APPROVED** [review]

**Pull request merged**